cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CaReconcilerTest.java

```diff
@@ -1451,4 +1459,233 @@ public void testClientsCASecretsWithoutOwnerReference(Vertx vertx, VertxTestCont
+                supplier, vertx, certManager, passwordGenerator);
+        mockCaReconciler
+                .reconcile(Clock.systemUTC())
+                .onComplete(context.succeeding(c -> context.verify(() -> {
+                    assertThat(mockCaReconciler.isClusterCaNeedFullTrust, is(true));
+                    assertThat(mockCaReconciler.zkPodRestartReasons.contains(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED), is(true));
+                    assertThat(mockCaReconciler.kPodRollReasons.contains(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED), is(true));
+                    assertThat(mockCaReconciler.deploymentRollReason.size() == 3, is(true));
+                    for (String reason: mockCaReconciler.deploymentRollReason) {
+                        assertThat(reason.equals(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED.getDefaultNote()), is(true));
+                    }
+                    async.flag();
+                })));
+    }
+
+    static class MockCaReconciler extends CaReconciler {
+
+        RestartReasons zkPodRestartReasons;
+        RestartReasons kPodRollReasons;
+        List<String> deploymentRollReason = new ArrayList<>();
+
+        public MockCaReconciler(Reconciliation reconciliation, Kafka kafkaCr, ClusterOperatorConfig config, ResourceOperatorSupplier supplier, Vertx vertx, CertManager certManager, PasswordGenerator passwordGenerator) {
+            super(reconciliation, kafkaCr, config, supplier, vertx, certManager, passwordGenerator);
+        }
+
+        @Override
+        Future<Void> verifyClusterCaFullyTrustedAndUsed() {
+            // assuming the CA key is not trusted
+            this.isClusterCaNeedFullTrust = true;
+            this.isClusterCaFullyUsed = false;
+            return Future.succeededFuture();
+        }
+
+        @Override
+        Future<Integer> getZooKeeperReplicas() {
+            return Future.succeededFuture(3);
+        }
+
+        @Override
+        Future<Void> maybeRollZookeeper(int replicas, RestartReasons podRestartReasons) {
+            this.zkPodRestartReasons = podRestartReasons;
+            return Future.succeededFuture();
+        }
+
+        @Override
+        Future<Set<NodeRef>> getKafkaReplicas() {
+            return Future.succeededFuture(KafkaCluster.nodes(reconciliation.name(), 3));
+        }
+
+        @Override
+        Future<Void> rollKafkaBrokers(Set<NodeRef> nodes, RestartReasons podRollReasons) {
+            this.kPodRollReasons = podRollReasons;
+            return Future.succeededFuture();
+        }
+
+        @Override
+        Future<Void> rollDeploymentIfExists(String deploymentName, String reason) {
+            return deploymentOperator.getAsync(reconciliation.namespace(), deploymentName)
+                    .compose(dep -> {
+                        if (dep != null) {
+                            this.deploymentRollReason.add(reason);
+                        }
+                        return Future.succeededFuture();
+                    });
```

Comment:
Wouldn't doing this through Mockito as the other operations be more elegant than doing it here in the override?

Reply:
The `deploymentOperator.getAsync` is already mocked in the test to return the Deployment(s).
What I am adding here is the rolling reason which is a field in the MockCaReconciler (and via the override I am avoiding the base class behaviour which would rolling the deployment instead).